acceptance/new-user/sign-up.test.js

```diff
@@ -1,105 +1,115 @@
-const puppeteer = require('puppeteer');
-const config = require('config');
+const puppeteer = require("puppeteer");
+const config = require("config");
 
-const URLS = require('../../constants/urls');
+const URLS = require("../../constants/urls");
 
 const { HOME_PAGE, SIGN_UP_PAGE, MY_HOST } = URLS;
 
 let browser, page;
 
+function delay(time) {
+    return new Promise(function (res) {
+        setTimeout(res, time);
+    });
+}
+
 beforeAll(async () => {
-  browser = await puppeteer.launch({
-    // headless: false,
-    // slowMo: 250
-  });
-
-  const context = await browser.createIncognitoBrowserContext();
-
-  page = await context.newPage();
-  // await page.emulate(puppeteer.devices['iPhone 8']);
-  // await page.setViewport({
-  //   width: 1440,
-  //   height: 900,
-  //   deviceScaleFactor: 1,
-  // });
-  return page;
+    browser = await puppeteer.launch({
+        headless: false,
+        slowMo: 0,
+    });
+
+    const context = await browser.createIncognitoBrowserContext();
+
+    page = await context.newPage();
+    // await page.emulate(puppeteer.devices['iPhone 8']);
+    // await page.setViewport({
+    //   width: 1440,
+    //   height: 900,
+    //   deviceScaleFactor: 1,
+    // });
+    return page;
 });
 
 afterAll(async () => {
-  await browser.close();
+    await browser.close();
 });
 
+describe("New user navigates", () => {
+    test("New user sess a sign up page", async () => {
+        await page.goto(HOME_PAGE);
+
+        await page.waitForResponse((res) => res.url().endsWith("/users/self"));
+        await page.waitForSelector("button.login-btn-text");
```

Comment:
_Doubt:_

`button.login-btn-text`
Is this a class name  ?
Which naming convention are we using here ?

Reply:
So this `waitForSelector` is a method in puppeteer which returns a promise when the particular selector(class, id, etc.) appears on the page. Now this particular test is for when a new member clicks the `sign up with github` button on our website and in our source code of the website, that particular button has the class `login-btn-text` . Now suppose we have multiple elements with the same class name, so we can specify that we need the `button` element with this particular class name and hence it says `button.login-btn-text`. The dot is for specifying that its a class.